File: Satori-iOS/Satori/Core/Theme/SatoriTheme.swift
import SwiftUI

// MARK: - 全局主题常量

struct SatoriTheme {
    // ── 圆角 ──
    static let cornerSmall:  CGFloat = 8
    static let cornerMedium: CGFloat = 16
    static let cornerLarge:  CGFloat = 24
    static let cornerPill:   CGFloat = 9999   // 胶囊形

    // ── 间距 ──
    static let spacingXS:  CGFloat = 4
    static let spacingS:   CGFloat = 8
    static let spacingM:   CGFloat = 16
    static let spacingL:   CGFloat = 24
    static let spacingXL:  CGFloat = 32
    static let spacingXXL: CGFloat = 48

    // ── 动画时长 ──
    static let animFast:   Double = 0.15
    static let animNormal: Double = 0.3
    static let animSlow:   Double = 0.6

    // ── 模糊 ──
    static let blurLight:  CGFloat = 20
    static let blurHeavy:  CGFloat = 40

    // ── 阴影 ──
    static let shadowSubtle = SatoriShadow(color: .black.opacity(0.06), radius: 8, y: 2)
    static let shadowMedium = SatoriShadow(color: .black.opacity(0.12), radius: 16, y: 4)
}

struct SatoriShadow {
    let color: Color
    let radius: CGFloat
    let y: CGFloat
}

// MARK: - View Modifier: 阴影

extension View {
    func satoriShadow(_ shadow: SatoriShadow) -> some View {
        self.shadow(color: shadow.color, radius: shadow.radius, x: 0, y: shadow.y)
    }
}

File: satori_flutter/lib/core/theme/satori_theme.dart
import 'package:flutter/material.dart';

/// 全局主题常量
class SatoriTheme {
  SatoriTheme._();

  // ── 圆角 ──
  static const cornerSmall = 8.0;
  static const cornerMedium = 16.0;
  static const cornerLarge = 24.0;
  static const cornerPill = 9999.0;

  // ── 间距 ──
  static const spacingXS = 4.0;
  static const spacingS = 8.0;
  static const spacingM = 16.0;
  static const spacingL = 24.0;
  static const spacingXL = 32.0;
  static const spacingXXL = 48.0;

  // ── 动画时长 ──
  static const animFast = Duration(milliseconds: 150);
  static const animNormal = Duration(milliseconds: 300);
  static const animSlow = Duration(milliseconds: 600);

  // ── 模糊 ──
  static const blurLight = 20.0;
  static const blurHeavy = 40.0;

  // ── 阴影 ──
  static const shadowSubtle = BoxShadow(
    color: Color(0x0F000000), // 0.06
    blurRadius: 8,
    offset: Offset(0, 2),
  );

  static const shadowMedium = BoxShadow(
    color: Color(0x1F000000), // 0.12
    blurRadius: 16,
    offset: Offset(0, 4),
  );
}
